Documentation/CRDs/Cluster/external-cluster/topology-for-external-mode.md

```diff
@@ -0,0 +1,201 @@
+# Topology provisioning for external mode
+
+## Background
+For Block storage (RWO), Enable Topology awareness for PV provisioning. Preferably, picking the replica 1 Storage Class for this AZ or creating a volume in a relevant AZ using a single storage class. It will help in providing Isolation to application/pods in AZ
+
+## Usage
+Applications like Kafka will have a deployment with number of instances. When using more than one instances, each service instances will create a new claim and will be located in a different zone. The configuration will include PVC and storage class. We will need to use replica 1 storage class and we will have PVC equivalent to instances, each PV is expected in a different zone.
+
+## Technical implementation details
+
+Add --topology-pools, --topology-failure-domain-label, --topology-failure-domain-values  to the python script.
+where,
+`--topology-pools`: (optional) comma-separated list of topology-constrained rbd pools
+`--topology-failure-domain-label`: (optional) k8s cluster failure domain label (example: zone,rack,host,etc) for the topology-pools that are matching the ceph domain
+ `--topology-failure-domain-values`: (optional) comma-separated list of the k8s cluster failure domain values corresponding to each of the pools in the topology-pools list
+
+And import script will create a new storage class `ceph-rbd-topology` if all these flags are passed during the export of ceph cluster data.
+
+## Example Steps Configuration and flow
+
+### RHCS cluster
+
+Looking at the osd tree, before jumping to the steps
```

Comment:
Please review the example for the following details:
- Don't use "minikube" for the host name. You need at least three unique host and zone names for the example.
- Don't include all the verbose ceph output unless it's really necessary to show a point. The ceph admins will understand the instructions if you just show the commands. Then we don't have to maintain the output in our docs.
- Use example pool names like `pool-a`, `pool-b`. The double letter pool names are confusing.
- Use example zone names like `zone-a`, `zone-b`, etc. It's nice for the example if the zone name corresponds to the pool names.

Reply:
>Don't include all the verbose ceph output unless it's really necessary to show a point. The ceph admins will understand the instructions if you just show the commands. Then we don't have to maintain the output in our docs.

Mostly kept there as to see the hierarchy and understand how the flow works for a particular hierarchy,  I can mention this is a specific cluster details it can very from user to user

Reply:
> Don't use "minikube" for the host name. You need at least three unique host and zone names for the example.

I think we use minikube in other docs too it would be fine, otherwise my whole initial flow would be not of use where I list nodes and all

>Use example zone names like zone-a, zone-b, etc. It's nice for the example if the zone name corresponds to the pool names.

We dont discuss about zones in this flow, its in the host

Reply:
Users who are using topology in their configuration will likely have zones, and not just hosts. I still suggest to use zones so the scenario makes more sense. You can modify the example output to have made-up zone names, then use pool names that are similar to the zone names. For example, `pool-a` belongs to `zone-a`.

Reply:
done